From 77d80edcb65316d95f8f09810613ee857dc7f8f9 Mon Sep 17 00:00:00 2001
From: Kemeng Shi <shikemeng@huawei.com>
Date: Mon, 24 May 2021 20:24:18 +0800
Subject: [PATCH 37/50] correct max_threads when max_threads is 0

Signed-off-by: Kemeng Shi <shikemeng@huawei.com>
---
 src/etmemd_src/etmemd_task.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/etmemd_src/etmemd_task.c b/src/etmemd_src/etmemd_task.c
index 01491f7..5aa693b 100644
--- a/src/etmemd_src/etmemd_task.c
+++ b/src/etmemd_src/etmemd_task.c
@@ -444,7 +444,7 @@ static int fill_task_threads(void *obj, void *val)
     int max_threads = parse_to_int(val);
     int core;
 
-    if (max_threads < 0) {
+    if (max_threads <= 0) {
         etmemd_log(ETMEMD_LOG_WARN,
                    "Thread count is abnormal, set the default minimum of current thread count to 1\n");
         max_threads = 1;
-- 
2.27.0

