From d99225f7e64d824b18c95fb475191ea50861169f Mon Sep 17 00:00:00 2001
From: Kemeng Shi <shikemeng@huawei.com>
Date: Mon, 24 May 2021 21:17:47 +0800
Subject: [PATCH 38/50] fix etmem help return error

Signed-off-by: Kemeng Shi <shikemeng@huawei.com>
---
 src/etmem_src/etmem.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/etmem_src/etmem.c b/src/etmem_src/etmem.c
index aadde74..76ea8d2 100644
--- a/src/etmem_src/etmem.c
+++ b/src/etmem_src/etmem.c
@@ -127,6 +127,7 @@ int main(int argc, char *argv[])
         if (conf.obj != NULL && strcmp(conf.obj, "help") == 0 &&
             argc == 2) { /* 2 is for param num of "etmem help" */
             err = 0;
+            goto out;
         }
         err = -EINVAL;
         goto out;
-- 
2.27.0

