From 10accbfc9c4d0a7658a333d94f5c3d0f31aeb6e5 Mon Sep 17 00:00:00 2001
From: Kemeng Shi <shikemeng@huawei.com>
Date: Wed, 26 May 2021 09:15:32 +0800
Subject: [PATCH 39/50] check if eng_mgt_func is NULL before use it

Signed-off-by: Kemeng Shi <shikemeng@huawei.com>
---
 src/etmemd_src/etmemd_project.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/etmemd_src/etmemd_project.c b/src/etmemd_src/etmemd_project.c
index 2e69dd6..decae76 100644
--- a/src/etmemd_src/etmemd_project.c
+++ b/src/etmemd_src/etmemd_project.c
@@ -732,6 +732,12 @@ enum opt_result etmemd_project_mgt_engine(const char *project_name, const char *
             return OPT_TASK_NOEXIST;
         }
     }
+
+    if (eng->ops->eng_mgt_func == NULL) {
+        etmemd_log(ETMEMD_LOG_ERR, "engine %s does not support eng_mgt_func\n", eng->name);
+        return OPT_INVAL;
+    }
+
     if (eng->ops->eng_mgt_func(eng, tk, cmd, sock_fd) != 0) {
         return OPT_INVAL;
     }
-- 
2.27.0

