From 1aef995d51254409b7aa1e2e84b9b750007e2413 Mon Sep 17 00:00:00 2001
From: YangXin <245051644@qq.com>
Date: Thu, 30 Sep 2021 18:58:33 +0800
Subject: [PATCH 50/50] Add CMakeLists.txt for three features of etmem.

Signed-off-by: YangXin <245051644@qq.com>
---
 CMakeLists.txt | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..a5d6761
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,18 @@
+# /******************************************************************************
+#  * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
+#  * etmem is licensed under the Mulan PSL v2.
+#  * You can use this software according to the terms and conditions of the Mulan PSL v2.
+#  * You may obtain a copy of Mulan PSL v2 at:
+#  *     http://license.coscl.org.cn/MulanPSL2
+#  * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR
+#  * IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR
+#  * PURPOSE.
+#  * See the Mulan PSL v2 for more details.
+#  * Author:YangXin
+#  * Create: 2021-09-31
+#  * Description:CMakeLists for three features of etmem to compile
+#  ******************************************************************************/
+
+add_subdirectory(etmem)
+add_subdirectory(memRouter)
+add_subdirectory(userswap)
\ No newline at end of file
-- 
2.27.0

