From 31731fdc3fbd8afd8daeba2138344568cbd6a238 Mon Sep 17 00:00:00 2001
From: liubo <liubo254@huawei.com>
Date: Tue, 21 Dec 2021 21:16:52 +0800
Subject: [PATCH 10/33] add make install support to CMakeList

---
 etmem/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/etmem/CMakeLists.txt b/etmem/CMakeLists.txt
index b5eb83e..9fba791 100644
--- a/etmem/CMakeLists.txt
+++ b/etmem/CMakeLists.txt
@@ -123,3 +123,12 @@ if( ${ARCHITECTURE} STREQUAL "aarch64" )
 else()
  target_compile_options(etmemd_scan PRIVATE -march=core-avx-i -m64)
 endif()
+
+install(TARGETS etmem etmemd 
+        PERMISSIONS OWNER_READ OWNER_EXECUTE 
+        RUNTIME DESTINATION /usr/bin)
+
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/conf/ 
+        DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE
+        FILE_PERMISSIONS OWNER_READ OWNER_WRITE 
+        DESTINATION /etc/etmem)
-- 
1.8.3.1

